schema.json

```diff
@@ -66,35 +66,65 @@
                                 "type" : "string"
                             },
                             "mode" : {
-                                "enum" : ["read", "write", "read-write"]
+                                "$ref": "#/$defs/AccessMode"
                             }
                         },
                         "additionalProperties" : false,
                         "required" : ["bucket", "mode"]
                     }
                 },
                 "mongodb": {
-                    "type": "boolean"
+                    "type": "array",
+                    "items" : {
+                        "type" : "object",
+                        "properties" : {
+                            "db" : {
+                                "type" : "string"
+                            },
+                            "mode" : {
+                                "$ref": "#/$defs/AccessMode"
+                            }
+                        }
+                    }
                 },
                 "vault" : {
-                    "type" : "boolean"
+                    "type" : "array",
+                    "items" : {
+                        "type" : "object",
+                        "properties" : {
+                            "path" : {
+                                "type" : "string"
+                            },
+                            "mode" : {
+                                "$ref": "#/$defs/AccessMode"
```

Comment:
Write-only is not something that we currently have a use-case for.

Reply:
Fixed access modes